### PR TITLE
Add .count()

### LINKED ISSUE #4: .count()
Add support for eg. `User.objects.count()`

### PR DESCRIPTION
**Pull request by florimondmanca**
Fixes #4 

Worth noting that `User.objects.all().count()` is not supported because `.all()` returns a list instead of a `QuerySet`. May seem counter-intuitive to people coming from Django.

### CODE REVIEW

File: orm/models.py [hunk 1]
```diff
@@ -156,13 +156,22 @@ def select_related(self, related):
             select_related=related,
         )
 
+    async def count(self) -> int:
+        expr = self.build_select_expression()
+        expr = sqlalchemy.func.count().select().select_from(expr)
+        count, = await self.database.fetch_one(expr)
```
Comment (tomchristie) [inline orm/models.py]: Same question as in the other PR - what's the unpacking?

File: orm/models.py [hunk 2]
```diff
@@ -110,13 +110,13 @@ def filter(self, **kwargs):
...
                 if related_parts:
-                    # Add any implied select_related
+                    #  Add any implied select_related
                     related_str = "__".join(related_parts)
                     if related_str not in select_related:
                         select_related.append(related_str)
 
                     # Walk the relationships to the actual model class
-                    # against which the comparison is being made.
+                    #  against which the comparison is being made.
```
Comment (tomchristie) [inline orm/models.py]: These whitespaces look unnecessary.

**Comment by tomchristie** [comment]
Really nice!

File: orm/models.py (hunk 2 above)
Comment (florimondmanca) [reply orm/models.py]: Yes! Black reformatted the code in all 3 PRs, and for some reason it added an extra space there which I forgot to remove here.

**Comment by tomchristie** [comment]
Same comment as #17 - databases 0.2.1 supports `fetch_val` - let's switch to using that then get this merged in.  👍

**Pull request merged**